src/solvers/td3.ml

```diff
@@ -580,7 +582,14 @@ module WP =
           let w = HM.find_default side_dep x VS.empty in
           HM.remove side_dep x;
 
-          if not (VS.is_empty w) && (not restart_only_globals || Node.equal (S.Var.node x) (Function Cil.dummyFunDec)) && not (S.Var.is_write_only x) then (
+          let should_restart =
+            if restart_only_access then
+              S.Var.is_write_only x
```

Comment:
The `only_access` name is a bit misleading here, since it's not particular to accesses (it doesn't try to crudely look into the unknowns, unlike the globals vs locals hack below). But the name `only_write_only` would also be very confusing.